tests/modules/sightings/resources/test_create_sighting.py

```diff
@@ -0,0 +1,94 @@
+
+    # default data_in will fail (no encounters)
+    response = sighting_utils.create_sighting(
+        flask_app_client, researcher_1, expected_status_code=400
+    )
+    assert response.json['passed_message'] == 'Must have at least one encounter'
+    assert not response.json['success']
+
+    # has encounters, zero assetReferences, but fails on bad taxonomy
+    data_in = {'encounters': [{'taxonomy': {'id': '0000000'}}]}
+    response = sighting_utils.create_sighting(
+        flask_app_client, researcher_1, expected_status_code=400, data_in=data_in
+    )
+    assert 'invalid taxonomy' in response.json['passed_message']['details']
+    assert not response.json['success']
+
+    # has encounters, but bunk assetReferences
+    data_in = {'encounters': [{'assetReferences': [{'fail': 'fail'}]}]}
+    response = sighting_utils.create_sighting(
+        flask_app_client, researcher_1, expected_status_code=400, data_in=data_in
+    )
+    assert (
+        response.json['passed_message'] == 'Invalid assetReference data in encounter(s)'
+    )
+    assert not response.json['success']
+
+    # assetReferences, but no files for them
+    data_in['encounters'][0]['assetReferences'][0] = {
+        'transactionId': transaction_id,
+        'path': 'i-dont-exist.jpg',
+    }
+    response = sighting_utils.create_sighting(
+        flask_app_client, researcher_1, expected_status_code=400, data_in=data_in
+    )
+    assert (
+        response.json['passed_message'] == 'Invalid assetReference data in encounter(s)'
+    )
+    assert not response.json['success']
+
+
+def test_create_and_delete_sighting(db, flask_app_client, researcher_1):
+    # from flask import current_app
+    from app.modules.sightings.models import Sighting
+    from app.modules.encounters.models import Encounter
+    import datetime
+    import os
+    import shutil
+
+    timestamp = datetime.datetime.now().isoformat()
+    transaction_id, test_filename = sighting_utils.prep_tus_dir()
+    data_in = {
+        'startTime': timestamp,
+        'encounters': [
+            {
+                'assetReferences': [
+                    {
+                        'transactionId': transaction_id,
+                        'path': test_filename,
+                    }
+                ]
+            }
+        ],
+    }
+    response = sighting_utils.create_sighting(
+        flask_app_client, researcher_1, expected_status_code=200, data_in=data_in
+    )
+    assert response.json['success']
+    sighting_id = response.json['result']['id']
+    encounter_id = response.json['result']['encounters'][0]['id']
+
+    # until DELETE api is completed, this is our cleanup  FIXME
+    encounter = Encounter.query.get(encounter_id)
+
+    # clean up submission
+    sub = encounter.get_assets()[0].submission
+    # encounter.delete_from_edm(current_app)
+    encounter.delete()
+
+    # now can delete sub since assets arent connected to enc
+    if os.path.exists(sub.get_absolute_path()):
```

Comment:
is it me, or is this a thing we should do upon `submission.delete()` by default?  is that too destructive?  alternately we could have the explicit `submission.delete_and_kill_directory()` or let it be default behavior and also have `submission.delete_leave_directory()`  or something.   :grin:

Reply:
Submission delete deletes all the assets. Asset delete deletes itself but not the file.
I don't see any benefit in keeping either the files or the directories once the Submission/Asset goes away

Reply:
kinda what i am feeling.  by then weve gone to a lot of trouble getting rid of stuff.   the only "benefit" is there is a chance that users potentially went to a lot of trouble to send us a lot of files from who knows what remote region of the world (e.g. for a bulk import)...  so i could see leaning toward not.  but that seems a little overly cautious.

Reply:
resolving this - but made jira ticket DEX-212 to remember this.